twitter/tweetwarc.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+"""
+Create WARC files from tweets stored in Apache Kafka in user defined dir.
+Read configuration from YAML file. Run like this:
+    tweetwarc.py -c tweetwarc.yaml -d /dir-to-save-warc
+"""
+from __future__ import unicode_literals
+import argparse
+import yaml
+import hashlib
+import uuid
+import json
+import logging
+import os
+from datetime import datetime
+from time import time
+from hanzo.warctools import WarcRecord
+from hanzo.warctools.warc import warc_datetime_str
+from kafka import KafkaConsumer
+
+logging.basicConfig(level=logging.INFO)
+
+
+def warc_uuid(text):
+    """Utility method for WARC header field urn:uuid"""
+    return ("<urn:uuid:%s>" %
+            uuid.UUID(hashlib.sha1(text).hexdigest()[0:32])).encode('ascii')
+
+
+def warc_filename(directory):
+    """WARC filename example: /tmp/tweets20170307100027.warc.gz.open
+    After the file is closed, remove the .open suffix
+    """
+    return "%s/tweets%s.warc.gz.open" % (
+        directory, datetime.utcnow().strftime('%Y%m%d%H%M%S'))
+
+def warcinfo_record(warc_filename):
+    """Return warcinfo WarcRecord.
+    Required to write in the beginning of a WARC file.
+    """
+    warc_date = warc_datetime_str(datetime.now())
+    metadata = "\r\n".join((
+        "format: WARC File Format 1.0",
+        "conformsTo: http://bibnum.bnf.fr/WARC/WARC_ISO_28500_version1_latestdraft.pdf"
+    ))
+    return WarcRecord(
+        headers=[
+            (WarcRecord.TYPE, WarcRecord.WARCINFO),
+            (WarcRecord.CONTENT_TYPE, b'application/warc-fields'),
+            (WarcRecord.ID, warc_uuid(metadata+warc_date)),
+            (WarcRecord.DATE, warc_date),
+            (WarcRecord.FILENAME, warc_filename)
+        ],
+        content=(b'application/warc-fields', metadata),
```

Comment:
I looked into `warctools.warc.WarcRecord` on how headers and content are written.
https://github.com/internetarchive/warctools/blob/master/hanzo/warctools/warc.py#L76
There is a common `nl` variable and after all headers are written we have 2 nl before starting the main content.

`WarcRecord` extends `ArchiveRecord` and the method `write_to` has the default value: `newline=b'\x0D\x0A'` which means `\r\n`
https://github.com/internetarchive/warctools/blob/master/hanzo/warctools/record.py#L167

So, `CRLF` seems to be already there.

To confirm, If you open an output warc file with vim, it is loaded in "dos mode" because the line endings are CRLF.

Also, If we appended a CRLF as you suggested, we'd have an extra empty line.